### PR TITLE
Fix union-type handling in PublicStaticPropertyFetchCollector

### PR DESCRIPTION
**Pull request by staabm**
analog https://github.com/TomasVotruba/unused-public/pull/120

### CODE REVIEW

File: src/Collectors/PublicStaticPropertyFetchCollector.php [hunk 1]
```diff
@@ -33,26 +36,38 @@ public function getNodeType(): string
...
+            && ($node->class->toString() === 'self' || $node->class->toString() === 'static')
+        ) {
+            // self fetch is allowed
             return null;
         }
 
-        if ($node->class->toString() === 'self') {
-            // self fetch is allowed
+        $classReflection = $scope->getClassReflection();
+        if ($classReflection instanceof ClassReflection && $this->classTypeDetector->isTestClass($classReflection)) {
```
Comment (staabm) [inline src/Collectors/PublicStaticPropertyFetchCollector.php:56]: this collector was missing the "test-only" detection part

**Comment by staabm** [comment]
should be good to go

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thanks 😊👍

**Comment by staabm** [comment]
would be great to get a new release. I have no further known problems to work on :-)

**Pull request merged**